### ISSUE TITLE
SWE Common JSON Encoding in Connected systems RFC

### ISSUE THREAD

**Issue by joanma747** [opengeospatial/ogcapi-connected-systems#71]
The basic example for the DataRecord on page 91 includes the values of the fields within the fields, not really useful

The example for a DataRecord inside a DataArray on page 96 does have a separate values block for the entire DataArray separate from the fields, and the encoding here is for an array of objects with key/value pairs, with the keys being the names of the fields in the DataRecord. The fields are ordered, as described as an array, so this could have a far more compact encoding (just arrays)

Going to the example of a TextEncoding on page 73 of a DataStream containing DataRecords, the encoding is efficient, not using key/value pairs (but is not an array, just a long text string that you have to parse yourself using the information from the TextEncoding. 

The example on page 100 does not provide actual data encoding, does provide an interestingly nested set of fields (3 on the base level, 2 of which are vectors themselves), no idea how they propose to flatten this

**Comment by alexrobin** [comment]
@KathiSchleidt @joanma747 We had such alternative encoding at some point in the early draft. We'll discuss adding this back in during our next telecon. I also think it's useful.

**Comment by alexrobin** [comment]
Some ideas for the encoding rules:
- If the DataRecord is itself within a DataArray, default to the nested JSON array encoding (like in @KathiSchleidt example)
- If not, default to JSON object encoding (I think we definitely need to keep the JSON object encoding option)
- Inside the JSONEncoding settings, allow to set the encoding type on a per record basis, which would override the defaults.

**Comment by KathiSchleidt** [comment]
@alexrobin I worry a bit about your cascade of subtleties guiding the encoding, would feel safer if we could just indicate the encoding utilized through an encodingType attribute. You already support textEncoding, why not introduce a jsonEncoding to clarify?

**Comment by alexrobin** [comment]
Discussed during 10/17 telecon.

Ok to add this feature in the JSON encoding. The idea would be to add a flag on the JSONEncoding object to turn the encoding of DataRecord as JSON arrays on or off.

**Comment by alexrobin** [comment]
@KathiSchleidt Yes, we decided to simplify the approach. We already have a `JSONEncoding` object. We will just add an attribute to indicate if DataRecords are encoded as objects or arrays.

**Comment by autermann** [comment]
How would we call this property? I though about `recordsAsArrays`, per default `false`. But wouldn't it make sense to also apply this to vectors? Just a generic `compact`? `objectsAsArrays`? `recordsAsArrays` and `vectorsAsArrays`?

**Comment by alexrobin** [comment]
Discussed during 12/12 telecon.

We agreed to use two separate flags: one for records called `recordsAsArrays`, one for vectors called `vectorsAsArrays`.
